From b8f3a7ccbeca5bdbd1b6210b94b38d3fef2dd0bd Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Thu, 19 Jan 2017 01:57:22 +0100
Subject: [PATCH 16/38] mtd: ubi: add auto_attach HACK for the ASUS RT-AC58U

This patch adds a hack that allows UBI's autoattach feature
to work with the custom ASUS UBI_DEV partition name.

This is necessary because the vendor's u-boot doesn't leave
the bootargs / cmdline alone, so the it can't be overwritten
easily otherwise.

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 drivers/mtd/ubi/build.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/mtd/ubi/build.c
+++ b/drivers/mtd/ubi/build.c
@@ -1170,6 +1170,9 @@ static void __init ubi_auto_attach(void)
 	mtd = open_mtd_device("ubi");
 	if (IS_ERR(mtd))
 		mtd = open_mtd_device("data");
+	/* Hack for the Asus RT-AC58U */
+	if (IS_ERR(mtd))
+		mtd = open_mtd_device("UBI_DEV");
 
 	if (!IS_ERR(mtd)) {
 		size_t len;
